### PR TITLE
New es

### PR DESCRIPTION
**Pull request by wguanicedew**
1) same like last pull request, except:
   a) move map between ddmendpoint and storage_id to infosys
   b) move storage keys to storagedata: tried to put get_security_key in infosys, but it infosys imports storagedata, if I import infosys in storagedata, it will be an error. If only put get_security_key in extinfo, it doesn't full make use of the infosys and it will make it even more complicated. So I keep get_security_key in storagedata currently(I know this part is not good).
2) I didn't test it. If there are no other objections, I will start to test it.

### CODE REVIEW

File: pilot/control/payloads/eventservice.py [hunk 1]
```diff
@@ -229,38 +32,47 @@ def run_payload(self, job, out, err):
         :param err:
         :return:
         """
-        log = get_logger(job.jobid)
+        log = get_logger(job.jobid, logger)
+
+        self.pre_setup(job)
+
+        self.post_setup(job)
```
Comment (PalNilsson) [inline pilot/control/payloads/eventservice.py]: This is not correct. Should be called after the setup has completed.

File: pilot/control/payloads/generic.py [hunk 2]
```diff
@@ -148,6 +156,91 @@ def run_payload(self, job, out, err):
...
+        :param out: (currently not used; deprecated)
+        :param err: (currently not used; deprecated)
+        :return: proc (subprocess returned by Popen())
+        """
+
+        log = get_logger(job.jobid, logger)
+
+        self.pre_setup(job)
+
+        self.post_setup(job)
```
Comment (PalNilsson) [inline pilot/control/payloads/generic.py]: Same comment as before, should be called after setup is done.

File: pilot/info/jobdata.py [hunk 3]
```diff
@@ -57,6 +58,7 @@ class JobData(BaseData):
     platform = ""   # cmtconfig value from the task definition
 
     is_eventservice = False        # True for event service jobs
+    is_eventservicemerge = False   # True for event service merge jobs
```
Comment (PalNilsson) [inline pilot/info/jobdata.py]: The version of this file is not the same as the one I committed yesterday. E.g. the 'status' field should now be a dictionary. Looking at the build number, you have build 19, but it should be 80..

File: pilot/common/pluginfactory.py [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+
+import logging
+logger = logging.getLogger(__name__)
+
+"""
+A factory to manage plugins
+"""
+
+
+class PluginFactory(object):
```
[the remark this answers is not in the document]
Comment (wguanicedew) [reply pilot/common/pluginfactory.py]: The code in pluginfactory is similar with this part.
The difference is that my part is using a class and other classes under pilot/eventservice inherit it. Then I don't need to write codes to call __import__ directly again. I just need to call the function which is already in the parent class.

File: pilot/control/payloads/eventservice.py (hunk 1 above)
Comment (wguanicedew) [reply pilot/control/payloads/eventservice.py]: updated.

File: pilot/control/payloads/generic.py (hunk 2 above)
Comment (wguanicedew) [reply pilot/control/payloads/generic.py]: updated.

File: pilot/info/jobdata.py (hunk 3 above)
Comment (wguanicedew) [reply pilot/info/jobdata.py]: rebased with your code.

**Comment by wguanicedew** [comment]
I rebased with your code. It seems there are no other objections. I will start to fix the bugs and test it with panda jobs.

File: pilot/info/jobdata.py (hunk 3 above)
Comment (wguanicedew) [reply pilot/info/jobdata.py]: I rebased with your code. Now status is dict.

**Comment by PalNilsson** [comment]
Thanks Wen. Note that I introduced a bug in the log creation at the end (the current build works with lsm-put but not with rucio upload when it tries to stage-out the log). I'll fix it asap (in the meantime you can take back the out-commented code in data::create_log() for the tar file usage if you want to test right away).

**Comment by PalNilsson** [comment]
In the meantime I have fixed the log issue. Will approve this, then merge with my latest dev code and create a new pull request.

**Pull request merged**